sdk/eventhub/Azure.Messaging.EventHubs/src/Compatibility/TrackOneGenericTokenProvider.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Messaging.EventHubs.Authorization;
+using Azure.Messaging.EventHubs.Core;
+using TrackOne;
+
+namespace Azure.Messaging.EventHubs.Compatibility
+{
+    /// <summary>
+    ///   A compatibility shim allowing a shared access signature to be used as a
+    ///   renewable security token with the Track One types.
+    /// </summary>
+    ///
+    /// <seealso cref="Authorization.EventHubTokenCredential"/>
+    /// <seealso cref="TrackOne.TokenProvider" />
+    ///
+    internal sealed class TrackOneGenericTokenProvider : TokenProvider
```

Comment:
Do we really want to call these types TrackOneXXX?

Reply:
Yes, we do.  These are short-term compatibility shims that will be removed after the first preview.  They exist only due to the short amount of time for delivering the preview.